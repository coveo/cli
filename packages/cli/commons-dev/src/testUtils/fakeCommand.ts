import {Command} from '@oclif/core';

export const getFakeCommand = (
  overrideConfig?: Record<string, unknown>
): Command => {
  const fakeCommand = {
    id: 'foo',
    warn: jest.fn(),
    log: jest.fn(),
    parse: jest.fn().mockReturnValue({flags: {}}),
    ...overrideConfig,
  };

  return fakeCommand as unknown as Command;
};
