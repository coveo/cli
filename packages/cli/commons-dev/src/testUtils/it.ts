import {fancy} from 'fancy-test';

/**
 * Wrapper around `jest.it` that captures all stdout/stderr.
 */
export const fancyIt = () => fancy.stdout().stderr().it;
