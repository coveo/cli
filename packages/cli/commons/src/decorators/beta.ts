import {CLICommand} from '../command/cliCommand';
import {DecoratorFunction} from './decoratorFunction';

export function Beta(): DecoratorFunction {
  return async function (target: CLICommand) {
    target.warn(
      `${target.identifier} is in beta, please report issue on https://github.com/coveo/cli/issues`
    );
  };
}
