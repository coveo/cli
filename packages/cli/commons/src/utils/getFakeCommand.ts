import {CLICommand} from '../command/cliCommand';

export const getFakeCommand = (
  overrideConfig?: Record<string, unknown>
): CLICommand => {
  const fakeCommand = {
    warn: jest.fn(),
    log: jest.fn(),
    parse: jest.fn().mockReturnValue({flags: {}}),
    identifier: 'foo',
    ...overrideConfig,
  };

  return fakeCommand as unknown as CLICommand;
};
