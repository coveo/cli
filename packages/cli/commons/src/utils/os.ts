export const appendCmdIfWindows = (cmd: TemplateStringsArray) =>
  `${cmd}${process.platform === 'win32' ? '.ps1' : ''}`;
