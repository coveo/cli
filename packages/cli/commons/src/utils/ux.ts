import {CliUx} from '@oclif/core';
import {red, green, magenta} from 'chalk';

export function startSpinner(task: string, status?: string) {
  if (CliUx.ux.action.running) {
    CliUx.ux.action.stop(green('✔'));
  }
  CliUx.ux.action.start(task, status);
}

export function stopSpinner(options?: {success?: boolean; message?: string}) {
  if (!CliUx.ux.action.running) {
    return;
  }
  const defaultOptions = {success: true, message: ''};
  const {success, message} = {...defaultOptions, ...options};
  const symbol = success ? green('✔') : red.bold('!');
  CliUx.ux.action.stop(`${symbol} ${message}`.trimEnd());
}

export const formatOrgId = (orgId: TemplateStringsArray | string) =>
  magenta(orgId);
