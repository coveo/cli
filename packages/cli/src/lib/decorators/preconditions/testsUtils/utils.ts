import {Command} from '@oclif/command';

export const getFakeCommand = (): Command => {
  const fakeCommand = {
    id: 'foo',
    warn: jest.fn(),
    log: jest.fn(),
    parse: jest.fn().mockReturnValue({flags: {}}),
  };

  return fakeCommand as unknown as Command;
};
