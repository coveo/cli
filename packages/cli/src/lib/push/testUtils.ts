export const doMockAxiosError = (
  status: number,
  message: string,
  errorCode: string
) => ({
  response: {
    status,
    data: {
      errorCode,
      message,
    },
  },
});

export const doMockAxiosSuccess = (status: number, text: string) => ({
  status,
  statusText: text,
});
