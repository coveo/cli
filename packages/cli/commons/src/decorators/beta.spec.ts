jest.mock('../platform/authenticatedClient');
jest.mock('../config/globalConfig');
jest.mock('../config/config');

import {fancyIt} from '@coveo/cli-commons-dev/testUtils/it';
import {getFakeCommand} from '../utils/getFakeCommand';
import {Beta} from './beta';

describe('beta', () => {
  const fakeCommand = getFakeCommand();
  jest.fn();
  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  fancyIt()(`warns that the command is in beta`, async () => {
    await Beta()(fakeCommand);
    expect(fakeCommand.warn).toBeCalledTimes(1);
    expect((fakeCommand.warn as jest.Mock).mock.lastCall).toMatchSnapshot();
  });
});
