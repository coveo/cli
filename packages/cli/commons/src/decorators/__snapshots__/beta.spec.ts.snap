// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`beta warns that the command is in beta 1`] = `
[
  "foo is in beta, please report issue on https://github.com/coveo/cli/issues",
]
`;
